### PR TITLE
Add dependencies checking and git pre-push hook

### PR DESCRIPTION
**Pull request by YuCJ**
Kanbanflow card:
[Add Jest test to ensure the shared dependencies will not cause duplicate dependencies be installed](https://kanbanflow.com/t/hhBgp47P)

### CODE REVIEW

File: dev/utils.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ * Build `dependencyVersionsAcrossPackages`. The returned object will be like:
+ * {
+ *   lodash: [ '^4.0.0', '^4.0.0', '^4.17.11', '^3.0.0' ],
+ *   'styled-components': [ '^4.0.0', '^4.0.0', '^4.0.0', '^4.0.0' ],
+ *   '@twreporter/core': [ '^1.0.1', '^1.0.3', '^1.0.3' ],
+ *   '@twreporter/velocity-react': [ '^1.4.1' ],
+ * }
+ *
+ * @export
+ * @param {string} [dependencyTypeShorthand] - none, 'dev', or 'peer'. As we use `yarn add` with the options.
```
Comment (nickhsine) [inline dev/utils.js]: `[dependencyTypeShorthand]` -> `dependencyTypeShorthand` ?

Comment (YuCJ) [reply dev/utils.js]: It's optional param. Like `yarn add xxx` means prod dependencies.
